webapp/components/integrations/components/edit_command/edit_command.jsx

```diff
@@ -4,147 +4,129 @@
+
+        /**
+        * Installed splash commands to display
+        */
+        commands: PropTypes.array,
+
+        /**
+        * The request state for editCommand action. Contains status and error
+        */
+        editCommandRequest: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+            * The function to call to fetch team commands
+            */
+            getCustomTeamCommands: PropTypes.func.isRequired,
+
+            /**
+            * The function to call to edit command
+            */
+            editCommand: PropTypes.func.isRequired
+        }).isRequired
     }
 
     constructor(props) {
         super(props);
 
-        this.handleIntegrationChange = this.handleIntegrationChange.bind(this);
-
-        this.submitCommand = this.submitCommand.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-        this.handleUpdate = this.handleUpdate.bind(this);
-        this.handleConfirmModal = this.handleConfirmModal.bind(this);
-        this.confirmModalDismissed = this.confirmModalDismissed.bind(this);
-
-        this.updateDisplayName = this.updateDisplayName.bind(this);
-        this.updateDescription = this.updateDescription.bind(this);
-        this.updateTrigger = this.updateTrigger.bind(this);
-        this.updateUrl = this.updateUrl.bind(this);
-        this.updateMethod = this.updateMethod.bind(this);
-        this.updateUsername = this.updateUsername.bind(this);
-        this.updateIconUrl = this.updateIconUrl.bind(this);
-        this.updateAutocomplete = this.updateAutocomplete.bind(this);
-        this.updateAutocompleteHint = this.updateAutocompleteHint.bind(this);
-        this.updateAutocompleteDescription = this.updateAutocompleteDescription.bind(this);
-
         this.originalCommand = null;
         this.newCommand = null;
 
-        const teamId = TeamStore.getCurrentId();
-
         this.state = {
-            displayName: '',
-            description: '',
-            trigger: '',
-            url: '',
-            method: REQUEST_POST,
-            username: '',
-            iconUrl: '',
-            autocomplete: false,
-            autocompleteHint: '',
-            autocompleteDescription: '',
             saving: false,
             serverError: '',
             clientError: null,
             showConfirmModal: false,
-            commands: IntegrationStore.getCommands(teamId),
-            loading: !IntegrationStore.hasReceivedCommands(teamId)
+            loading: true
         };
     }
 
     componentDidMount() {
-        IntegrationStore.addChangeListener(this.handleIntegrationChange);
-
         if (window.mm_config.EnableCommands === 'true') {
-            loadTeamCommands();
+            this.props.actions.getCustomTeamCommands(this.props.team.id).then(
+                () => {
+                    this.originalCommand = this.props.commands.filter((command) => command.id === this.props.commandId)[0];
```

Comment:
This is the old behavior. The question is - Could I rely on commands data from redux that was fetched when CommandsContainer did mount? Do I really need to fetch them all again every time I visit edit component?